js/adapt-assessmentArticleModel.js

```diff
@@ -121,7 +121,8 @@ const AssessmentModel = {
   _setupAssessmentData(force, callback) {
     const assessmentConfig = this.getConfig();
     const state = this.getState();
-    const shouldResetAssessment = (!this.get('_attemptInProgress') && !state.isPass) || force === true;
+    const hasAttemptsLeft = (state.attemptsLeft > 0 || state.attemptsLeft === 'infinite');
+    const shouldResetAssessment = (!this.get('_attemptInProgress') && !state.isPass && hasAttemptsLeft) || force === true;
```

Comment:
Need to check `_isResetOnRevisit` otherwise the assessment will still reset.

Reply:
Amended as part of https://github.com/adaptlearning/adapt-contrib-assessment/issues/155.